db/migrate/20150608190453_create_results.rb

```diff
@@ -0,0 +1,17 @@
+class CreateResults < ActiveRecord::Migration
+  def change
+    create_table :results do |t|
+      t.integer :assessment_id
+      t.string :assessment_type
+      t.string :name
+      t.string :description
+      t.string :problem_description
+      t.integer :percentage
+      t.integer :year
+      t.string :semester
+      t.timestamps null: false
+    end
+
+    add_index :results, [:assessment_id, :assessment_type]
```

Comment:
What do you think about adding:

```ruby
add_index :results, [:assessment_id, :assessment_type, :year, :semester], unique: true
```

The intention being to guard against duplicate results for the same period.